File: app/src/main/java/com/hjc/base/viewmodel/PageViewModel.java
package com.hjc.base.viewmodel;


import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;
import androidx.paging.DataSource;
import androidx.paging.LivePagedListBuilder;
import androidx.paging.PagedList;

import com.hjc.base.bean.ConcertBean;
import com.hjc.base.ui.home.page.ConcertFactory;
import com.hjc.baselib.viewmodel.CommonViewModel;

public class PageViewModel extends CommonViewModel {

    private final LiveData<PagedList<ConcertBean>> convertList;
    private DataSource<Integer, ConcertBean> concertDataSource;

    public PageViewModel(@NonNull Application application) {
        super(application);

        ConcertFactory concertFactory = new ConcertFactory();
        concertDataSource = concertFactory.create();
        convertList = new LivePagedListBuilder<>(concertFactory, 20).build();
    }

    public void invalidateDataSource() {
        concertDataSource.invalidate();
    }

    public LiveData<PagedList<ConcertBean>> getConvertList() {
        return convertList;
    }
}

File: module-home/src/main/java/com/hjc/module_home/viewmodel/paging/PageViewModel.kt
package com.hjc.module_home.viewmodel.paging

import android.app.Application
import androidx.lifecycle.LiveData
import androidx.paging.DataSource
import androidx.paging.LivePagedListBuilder
import androidx.paging.PagedList
import com.hjc.library_base.viewmodel.BaseViewModel
import com.hjc.module_home.http.entity.ConcertBean
import com.hjc.module_home.ui.paging.ConcertFactory

class PageViewModel(application: Application) : BaseViewModel(application) {

    private val concertDataSource: DataSource<Int, ConcertBean>

    val convertList: LiveData<PagedList<ConcertBean>>

    init {
        val concertFactory = ConcertFactory()
        concertDataSource = concertFactory.create()
        convertList = LivePagedListBuilder(concertFactory, 20).build()
    }

    fun invalidateDataSource() {
        concertDataSource.invalidate()
    }

}